tools/commit_slider/case_1.md

```diff
@@ -0,0 +1,316 @@
+    # populating of commit path
+    pathCommit.firstThroughput = self.firstThroughput
+    pathCommit.firstModel = self.firstModel
+    pathCommit.secondThroughput = self.secondThroughput
+    pathCommit.secondModel = self.secondModel
+
+def  getCommitInfo(self, commit):
+    return  "{hash}, throughput_1 = {t1}, throughput_2 = {t2}".format(
+        hash=commit.cHash,
+        t1=commit.firstThroughput,
+        t2=commit.secondThroughput)
+```
+### Configuration
+```
+supposedly we pass both commandlines launching models in two parameters cfg["runConfig"]["firstAppCmd"] and cfg["runConfig"]["secondAppCmd"], i.e.
+cfg = { 
+    "appCmd" : "{actualAppCmd}",
+    "appPath": "<path_to_benchmark_app>",
+    "gitPath" : "<path_to_git_repository>",
+    "buildPath" : "<path_to_build_dir>",
+        "runConfig" : {
+            "commitList" : {
+                "explicitList" : [ "{start}", "{end}" ]
+            },
+            "mode" : "crossCheck",
+            "firstAppCmd" : "./benchmark_app -m model_1.xml --other_params",
+            "secondAppCmd" : "./benchmark_app -m model_2.xml --other_params",
+            "traversal" : "bruteForce"
+        }
+}
+```
+### Output
+```
+python3 commit_slider.py -cfg cfg.json
+
+<hash_1>, throughput_1 = 500.0, throughput_2 = 500.0
+<hash_2>, throughput_1 = 1000.0, throughput_2 = 1000.0
+
+```
+We've got rid of full output and learned to provide only necessary information, but we still limited by verbose configuration-style API and still provide raw output information.
+
+## Third Version: Specific Template
+###  API design
+We want to provide for user 2 ways to set up command lines for comparation. ```'appCmd' : ['./benchmark_app --parameters_1', './benchmark_app --parameters_2']``` for detailed parameters and simplified ```'par_1' : 'parameter_1', 'par_2' : 'parameter_2'``` supposing that substitute the placeholder ```'appCmd' : 'prefix {actualPar} postfix'```. The example of this approach may be ```'par_1' : 'model_1.xml', 'par_2' : 'model_2.xml' : 'parameter_2', 'appCmd' : './benchmark_app -m {actualPar} -i input.png'```
+  
+### Custom template
+```
+from  utils.templates.common_template  import  Template
+
+class  BenchmarkCrossCheckTemplate(Template):
+
+def  printResult(commitPath, outLogger, getCommitInfo):
+    # formatting of output anf post analisys
+    print("\n" + "*" * 17 "<Commit slider output>" + "*" * 17 "\nTable of throughputs:" + '\n\t\thr_1\thr_2')
+    for  pathcommit  in  commitPath.getList():
+        commitInfo = getCommitInfo(pathcommit)
+        cHash = pathcommit.cHash
+        cHash = cHash[:7]
+        commMsg = "{}\t\t{}\t{}".format( cHash,
+            pathcommit.firstThroughput,
+            pathcommit.secondThroughput)
+        print(commMsg)
+        outLogger.info(commitInfo)
+    cPath = commitPath.getList()
+    thr00 = float(cPath[0].firstThroughput)
+    thr01 = float(cPath[0].secondThroughput)
+    thr10 = float(cPath[1].firstThroughput)
```

Comment:
is `thr11` missed?

Reply:
No, we print full table, but for resolving the problem issue we need only 3 variables.